miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/MiniAppSharedPreferences.kt

```diff
@@ -0,0 +1,16 @@
+package com.rakuten.tech.mobile.miniapp.storage
+
+import android.content.Context
+import android.content.SharedPreferences
+
+internal class MiniAppSharedPreferences(context: Context) {
+    private val prefs: SharedPreferences = context.getSharedPreferences(
+        "com.rakuten.tech.mobile.miniapp.storage", Context.MODE_PRIVATE
+    )
+
+    fun setAppExisted(appId: String, versionId: String, value: Boolean) =
+        prefs.edit().putBoolean(appId + versionId, value).apply()
```

Comment:
In the future, we will also need to know what version was downloaded because we will need to delete the previous version when a new version is downloaded for the specified App ID. So could we cache the versionId by AppId here, i.e. `prefs.edit().putString(appId, versionId)`. 

What do you think? Were you thinking of another way of handling deletion?

Reply:
I don't think that we need to store the specific appId and versionId here.
The remove of previous version occurs when the download of new version completes -> We know the appId and the new versionId so we can remove all older version folders.

Actually, I have already implemented that feature but not in this PR.

```MiniAppStorage.kt
private fun getParentPathApp(appId: String) = "${basePath.path}/$SUB_DIR_MINIAPP/$appId/"
```
```MiniAppDownloader.kt
fun removeOutdatedVersionApp(appId: String, versionId: String) {
    val parentFile = File(getParentPathApp(appId))

    if (parentFile.isDirectory && parentFile.listFiles() != null) {
        parentFile.listFiles().forEach {file ->
            if (!file.absolutePath.endsWith(versionId))
                file.deleteRecursively()
        }
    }
}
```

Reply:
Ah, okay nice solution! We just need to make sure that the old version isn't removed until after the new version has completed downloading (in case the new version download fails).